pkg/platform/packet/packet.go

```diff
@@ -239,6 +209,40 @@ func createTerraformConfigFile(cfg *config, terraformPath string) error {
 	return nil
 }
 
+func (c *config) terraformSmartApply(ex *terraform.Executor) error {
+	dnsProvider, err := dns.ParseDNS(&c.DNS)
```

Comment:
I would leave this function in the original place, as it's a "validation call" and not "action call" (like "ex.Apply" for example). This way, error returned from this function will always be a "runtime error".

Reply:
It also reduces number of `return` calls to 4, which is what Code climate suggests as maximum.